src/client/java/minicraft/screen/AppToast.java

```diff
@@ -0,0 +1,112 @@
+package minicraft.screen;
+
+import minicraft.gfx.Color;
+import minicraft.gfx.Dimension;
+import minicraft.gfx.Font;
+import minicraft.gfx.FontStyle;
+import minicraft.gfx.Insets;
+import minicraft.gfx.Screen;
+import minicraft.gfx.Sprite;
+import minicraft.gfx.SpriteLinker;
+import org.intellij.lang.annotations.MagicConstant;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class AppToast extends Toast {
+	private static final int TOP_PADDING = 30;
+	private static final int BOTTOM_PADDING = 30;
+	private static final int MAX_WIDTH = Screen.w * 3 / 8;
+	private static final int MAX_HEIGHT = Screen.w * 3 / 8;
+
+	@MagicConstant(intValues = {0, 1})
+	private static int positioning = 0;
+
+	public static void setPositionTop() {
+		positioning = 0;
+	}
+	public static void setPositionBottom() {
+		positioning = 1;
+	}
+
+	private final Dimension size;
+	private final List<String> text;
+	private final AppToastFrame frame;
+	private final int color;
+
+	public static abstract class AppToastFrame extends ToastFrame {
+		public static final AppToastFrame FRAME_GENERAL = new GeneralAppToastFrame();
+		public static final AppToastFrame FRAME_URGENT = new UrgentAppToastFrame();
+		public static final AppToastFrame FRAME_WINDOW = new WindowAppToastFrame();
+		public static final AppToastFrame FRAME_BRICK = new BrickAppToastFrame();
+
+		protected final @Nullable SpriteLinker.LinkedSprite sprite;
+
+		protected AppToastFrame(@Nullable SpriteLinker.LinkedSprite sprite, Insets paddings) {
+			super(paddings);
+			this.sprite = sprite;
+		}
+
+		@Override
+		public void render(Screen screen, int x, int y, int w, int h) {
+			assert sprite != null; // If sprite == null, this implementation should be overridden.
+			render(screen, x, y, w, h, sprite.getSprite());
+		}
+
+		private static class GeneralAppToastFrame extends AppToastFrame {
+			private GeneralAppToastFrame() {
+				super(new SpriteLinker.LinkedSprite(SpriteLinker.SpriteType.Gui, "toasts")
+					.setSpriteDim(0, 0, 3, 3), new Insets(4));
+			}
+		}
+
+		private static class UrgentAppToastFrame extends AppToastFrame {
+			private UrgentAppToastFrame() {
+				super(new SpriteLinker.LinkedSprite(SpriteLinker.SpriteType.Gui, "toasts")
+					.setSpriteDim(3, 0, 3, 3), new Insets(4));
+			}
+		}
+
+		private static class WindowAppToastFrame extends AppToastFrame {
+			private WindowAppToastFrame() {
+				super(new SpriteLinker.LinkedSprite(SpriteLinker.SpriteType.Gui, "toasts")
+					.setSpriteDim(0, 3, 3, 3), new Insets(6, 4, 2, 8));
+			}
+		}
+
+		private static class BrickAppToastFrame extends AppToastFrame {
```

Comment:
These don't need to be classes as they are just a function within a class that returns a value. Can be replaced by a variable.